### ISSUE TITLE
Many broken tests after recent Clang update on CentOS 8 Stream

### ISSUE THREAD

**Issue by hahnjo** [root-project/root#11601]
After updating to `clang-14.0.6-1.module_el8.7.0+1198+0c3eb6e2.x86_64`, many tests started failing when compiling ROOT with Clang (134 tests in my configuration). The reason seems to be that Clang is now built with [`GCC_INSTALL_PREFIX=/opt/rh/gcc-toolset-12/root/usr`](https://git.centos.org/rpms/clang/c/4e0495208f843a286dc0d06d9abdc3433a66983a) to detect and use `libstdc++` from GCC 12. Even if that was not the case, https://github.com/llvm/llvm-project/commit/35aaf548237a4f213ba9d95de53b33c5ce1eadce in Clang 15 will have a similar effect for people building the compiler from source.

According to our own documentation, ROOT does not guarantee ["compatibility with compilers shipped with `devtoolsets` on CentOS or Red Hat"](https://root.cern/install/build_from_source/#caveats) which right now only leaves building with the system-default GCC 8.5 to get a fully working installation of ROOT.

**Comment by hahnjo** [comment]
As a workaround, `cmake -DCLING_CXX_PATH=/usr/bin/g++` seems to work. I would have loved to add this as a tweak to the build system, but we have to address https://github.com/root-project/root/issues/11612 first...

**Comment by dpiparo** [comment]
Sorry for coming back to this after all these months: but is this still relevant?

**Comment by hahnjo** [comment]
Yes, the workaround is still needed for all Clang builds on EL8 (see for example also the ASan build configuration)

**Comment by hahnjo** [comment]
Final workaround, now that Clang supports configuration files:
```
$ cat /etc/clang/clang++.cfg
# Force clang++ to use the system-default GCC and libstdc++, not the GCC Toolset.
--gcc-install-dir=/usr/lib/gcc/x86_64-redhat-linux/8/
```

Closing.
